File: frontend/src/types/customer.ts
export interface Customer {
  id: number;
  name: string;
  email: string;
  shopify_store_url: string;
  business_name?: string;
  contact_person?: string;
  phone?: string;
  address?: string;
  is_active: boolean;
  webhook_enabled: boolean;
  auto_fulfillment: boolean;
  created_at: string;
  updated_at?: string;
  owner_id: number;
}

export interface CustomerCreate {
  name: string;
  email: string;
  shopify_store_url: string;
  shopify_access_token: string;
  printify_api_token: string;
  business_name?: string;
  contact_person?: string;
  phone?: string;
  address?: string;
  webhook_enabled?: boolean;
  auto_fulfillment?: boolean;
}

export interface CustomerUpdate extends Partial<CustomerCreate> {}

File: backend/app/schemas/customer.py
"""
Customer schemas
"""

from typing import Optional
from pydantic import BaseModel, EmailStr
from app.schemas.base import BaseResponse


class CustomerBase(BaseModel):
    name: str
    email: EmailStr
    shopify_store_url: str
    business_name: Optional[str] = None
    contact_person: Optional[str] = None
    phone: Optional[str] = None
    address: Optional[str] = None
    webhook_enabled: bool = True
    auto_fulfillment: bool = True


class CustomerCreate(CustomerBase):
    shopify_access_token: str
    shopify_api_secret: Optional[str] = None
    printify_shop_id: Optional[str] = None


class CustomerUpdate(BaseModel):
    name: Optional[str] = None
    email: Optional[EmailStr] = None
    shopify_store_url: Optional[str] = None
    business_name: Optional[str] = None
    contact_person: Optional[str] = None
    phone: Optional[str] = None
    address: Optional[str] = None
    webhook_enabled: Optional[bool] = None
    auto_fulfillment: Optional[bool] = None


class CustomerResponse(CustomerBase, BaseResponse):
    id: int
    is_active: bool
    owner_id: int
